packages/amazon-sumerian-hosts-babylon/src/Babylon.js/HostUtils.js

```diff
@@ -0,0 +1,578 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: MIT-0
+
+/**
+ * The HOST.HostUtils module provides helper functions for instantiating Sumerian Hosts.
+ * @namespace babylonjs/HostUtils
+ *
+ * @example
+const characterId = 'Cristine';
+const characterConfig = HOST.HostUtils.getCharacterConfig('./assets/character-assets', characterId);
+const pollyConfig = { pollyVoice: 'Joanna', pollyEngine: 'neural' };
+const host = await HOST.HostUtils.createHost(scene, characterConfig, pollyConfig);
+ */
+
+/* eslint-disable no-param-reassign */
+
+import { LipsyncFeature, GestureFeature } from '@amazon-sumerian-hosts/core';
+import anim from './animpack';
+import aws from './awspack';
+import HostObject from './HostObject';
+import PointOfInterestFeature from './PointOfInterestFeature';
+
+let isTextToSpeechInitialized = false;
+
+/**
+ * Creates a new Sumerian Host from the assets listed in the characterConfig
+ * parameter. This can be used to create one of the built-in hosts or your own
+ * custom host.
+ *
+ * When creating a custom host, use SumerianHostUtils.getCharacterConfig() to
+ * retrieve the appropriate config for that character.
+ *
+ * @memberof babylonjs/HostUtils
+ * @param {BABYLON.Scene} scene The scene to add the host to.
+ * @param {Object} characterConfig
+ * @param {String} characterConfig.modelUrl
+ * @param {String} characterConfig.animStandIdleUrl
+ * @param {String} characterConfig.animLipSyncUrl
+ * @param {String} characterConfig.animGestureUrl
+ * @param {String} characterConfig.animEmoteUrl
+ * @param {String} characterConfig.animFaceIdleUrl
+ * @param {String} characterConfig.animBlinkUrl
+ * @param {String} characterConfig.animPointOfInterestUrl
+ * @param {Object} pollyConfig
+ * @param {string} pollyConfig.pollyVoice The Polly voice to use. See
+ *   {@link https://docs.aws.amazon.com/polly/latest/dg/voicelist.html}
+ * @param {string} pollyConfig.pollyEngine The Polly engine you would like to
+ *   use. Either "standard" or "neural". Note that the neural engine incurs a
+ *   higher cost and is not compatible with all voices or regions. See
+ *   {@link https://docs.aws.amazon.com/polly/latest/dg/NTTS-main.html}
+ *
+ * @returns {babylonjs/HostObject} A functioning Sumerian Host
+ */
+async function createHost(scene, characterConfig, pollyConfig) {
+  await initTextToSpeech();
+  const assets = await loadAssets(scene, characterConfig);
+  const host = assembleHost(assets, scene);
+  addTextToSpeech(host, scene, pollyConfig.pollyVoice, pollyConfig.pollyEngine);
+  addPointOfInterestTracking(host, scene, assets.poiConfig);
+
+  return host;
+}
+
+/**
+ * @private
+ */
+async function initTextToSpeech() {
+  // Ensure services get initialized only once per session.
+  if (isTextToSpeechInitialized) return;
```

Comment:
Do we need this separate tracer variable or we could just use aws.TextToSpeechFeature.isReady

Reply:
I'll look into this. I had just kept the original implementation as it was and wasn't aware of the "isReady" property.

Reply:
This issue is addressed in the update I just posted.